src/App.tsx

```diff
@@ -1,15 +1,38 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { debounce } from 'lodash';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { Dropdown } from './components/Dropdown';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  // const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person>();
+  const [isVisible, setIsVisible] = useState(false);
+  const [query, setQuery] = useState('');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setQuery, 300), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsVisible(true);
+    setSelectedPerson(undefined);
+    applyQuery(event.target.value);
```

Comment:
What does this function do? You set to variable only last letter which I typed, but you should use debounce only for request (not for input)

https://github.com/mate-academy/react_autocomplete/assets/70009205/3f855b88-17f0-4448-a4f7-66c1cb405240